File: src/Female.java
import java.util.Objects;

public class Female extends Family {

    private String nameHusband;

    public Female(String firstName, String secondName, String nameFather, String nameMother, int age, String nameHusband) {
        super(firstName, secondName, nameFather, nameMother, age);
        this.nameHusband = nameHusband;
    }

    public Female(String firstName, String secondName, String nameFather, String nameMother, int age) {
        super(firstName, secondName, nameFather, nameMother, age);
    }

    public Female() {

    }

    public String getNameHusband() {
        return nameHusband;
    }

    public void setNameHusband(String nameHusband) {
        this.nameHusband = nameHusband;
    }

    @Override
    public String toString() {
        return "Female{" +
                "nameHusband='" + nameHusband + '\'' +
                "} " + super.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Female female = (Female) o;
        return nameHusband.equals(female.nameHusband);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), nameHusband);
    }
}

File: PythonOOP/FamilyPack/Female.py
from FamilyPack import MemberFamily


class Female(MemberFamily):
    def __init__(self, first_name, second_name, name_father, name_mother, age, name_husband):
        super().__init__(first_name, second_name, name_father, name_mother, age)
        self._name_husband = name_husband

    @property
    def name_husband(self):
        return self._name_husband

    @name_husband.setter
    def name_husband(self, value):
        self._name_husband = value

    def __str__(self):
        return super().__str__() + '\n' + str(self.name_husband)



